src/net/http/responsecontroller_test.go

```diff
@@ -263,3 +263,18 @@ func testWrappedResponseController(t *testing.T, mode testMode) {
 	io.Copy(io.Discard, res.Body)
 	defer res.Body.Close()
 }
+
+func TestIssue58237(t *testing.T) {
+	cst := newClientServerTest(t, http2Mode, HandlerFunc(func(w ResponseWriter, req *Request) {
+		ctl := NewResponseController(w)
+		if err := ctl.SetReadDeadline(time.Now().Add(1 * time.Millisecond)); err != nil {
+			t.Errorf("ctl.SetReadDeadline() = %v, want nil", err)
+		}
+		time.Sleep(10 * time.Millisecond)
```

Comment:
since the `1ms` and `10ms` are related, my I suggest:
```suggestion
	cst := newClientServerTest(t, http2Mode, HandlerFunc(func(w ResponseWriter, req *Request) {
		ctl := NewResponseController(w)
		duration := 1 * time.Millisecond
		if err := ctl.SetReadDeadline(time.Now().Add(duration)); err != nil {
			t.Errorf("ctl.SetReadDeadline() = %v, want nil", err)
		}
		time.Sleep(10 * duration)
```